core-to-plc/test/Spec.hs

```diff
@@ -9,26 +9,40 @@ module Main (main) where
 import           IllTyped
 
 import           Language.Plutus.CoreToPLC.Plugin
-import           Language.Plutus.CoreToPLC.Primitives as Prims
+import qualified Language.Plutus.CoreToPLC.Primitives     as Prims
 
 import           Language.PlutusCore
+import           Language.PlutusCore.Evaluation.CkMachine
 
 import           Test.Tasty
 import           Test.Tasty.Golden
 
 import           Control.Exception
-import qualified Data.ByteString.Lazy                 as BSL
-import           Data.Text                            as T
-import           Data.Text.Encoding                   (encodeUtf8)
+import           Control.Monad.Except                     hiding (void)
+
+import qualified Data.ByteString.Lazy                     as BSL
+import qualified Data.Text                                as T
+import           Data.Text.Encoding                       (encodeUtf8)
 
 -- this module does lots of weird stuff deliberately
 {-# ANN module "HLint: ignore" #-}
 
 main :: IO ()
 main = defaultMain tests
 
+getPlc :: PlcCode -> ExceptT BSL.ByteString IO (Program TyName Name ())
+getPlc value = withExceptT (strToBs . show) $ getAst <$> (ExceptT $ try @SomeException (evaluate value))
```

Comment:
This should probably be in `Language.Plutus.CoreToPLC.Plugin`, where `getAst` is defined

Reply:
I don't think so - the user isn't really supposed to know that it can throw, that should only happen when you're running with `defer-errors` which is only really there for testing.

Reply:
Plus I don't really want to try and make the exception handling production-quality.